src/modules/action-button/action-button-link.ts

```diff
@@ -0,0 +1,6 @@
+import { StacheNavLink } from '../nav/nav-link';
+
+export interface StacheActionButtonLink extends StacheNavLink {
```

Comment:
While I think this approach is best for the Action Button component, we may be missing an opportunity to use the icon and summary for other components that may require it. For example, we may (at some point) want to create another nav component that utilizes an icon or a summary. It might be better to adjust the `StacheNavLink` interface, instead of creating a new one for this component.

```
export interface StacheNavLink {
  name: string;
  path: string[] | string;
  fragment?: string;
  icon?: string;
  summary?: string;
}
```

I could also see a use for this on the breadcrumbs component: changing "Home" to use a house icon...